pkg/comp-functions/functions/common/instance_namespace.go

```diff
@@ -93,6 +94,19 @@ func createInstanceNamespace(ctx context.Context, serviceName, compName, claimNa
 		},
 	}
 
+	controlNS, ok := svc.Config.Data["controlNamespace"]
+	if !ok {
+		return fmt.Errorf("controlNamespace not specified")
+	}
+
+	disabled, err := isBillingDisabled(controlNS, instanceNamespace, compName, svc)
+	if err != nil {
+		return err
+	}
+	if disabled {
+		ns.ObjectMeta.Labels["appuio.io/billing-name"] = ""
```

Comment:
Are we sure the label will not be changed in the future? It would be a good idea to dynamically get these labels if possible instead of hard coding them here.

Reply:
We fully control the queries for the billing ourselves, see here: https://github.com/vshn/component-appcat/tree/master/component/component/promql

Chaning these will need a lot of consideration, planing and testing, so I don't think dynamically passing them dynamically would really give a big benefit.